src/Nph/Order/TimePoints.php

```diff
@@ -81,4 +81,19 @@ public function getSamplesByType($type): array
         }
         return $samples;
     }
+
+    public function getSampleLabelFromCode(string $sampleCode): string
+    {
+        $samplesInfo = $this->getSamplesInformation();
+        return $samplesInfo[$sampleCode]['label'];
+    }
+
+    public function getSampleCollectionVolumeFromCode(string $sampleCode): string
```

Comment:
I refactored the code and moved all sample specific methods in to Samples class. Could you move these two  methods to Sample class? If you pull my recent branch `sn/hpro-1255/nph-order-write-test-cases` you should see those changes.